cmd/podman/volumes/import.go

```diff
@@ -0,0 +1,75 @@
+package volumes
+
+import (
+	"context"
+	"os"
+
+	"github.com/containers/podman/v3/cmd/podman/common"
+	"github.com/containers/podman/v3/cmd/podman/inspect"
+	"github.com/containers/podman/v3/cmd/podman/parse"
+	"github.com/containers/podman/v3/cmd/podman/registry"
+	"github.com/containers/podman/v3/pkg/domain/entities"
+	"github.com/containers/podman/v3/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	importDescription = `Imports contents into a podman volume from specified tarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz).`
+	importCommand     = &cobra.Command{
+		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Use:               "import VOLUME [SOURCE]",
+		Short:             "Import a tarball contents into a podman volume",
+		Long:              importDescription,
+		RunE:              importVol,
+		Args:              cobra.ExactArgs(2),
+		ValidArgsFunction: common.AutocompleteVolumes,
+		Example: `podman volume import my_vol /home/user/import.tar
+  cat ctr.tar | podman import volume my_vol -`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: importCommand,
+		Parent:  volumeCmd,
+	})
+}
+
+func importVol(cmd *cobra.Command, args []string) error {
+	var inspectOpts entities.InspectOptions
+	var tarFile *os.File
+	containerEngine := registry.ContainerEngine()
+	ctx := context.Background()
+	// create a slice of volumes since inspect expects slice as arg
+	volumes := []string{args[0]}
+	tarPath := args[1]
+
+	if tarPath != "-" {
+		err := parse.ValidateFileName(tarPath)
+		if err != nil {
+			return err
+		}
+
+		// open tar file
+		tarFile, err = os.Open(tarPath)
+		if err != nil {
+			return err
+		}
+	} else {
+		tarFile = os.Stdin
+	}
+
+	inspectOpts.Type = inspect.VolumeType
+	volumeData, _, err := containerEngine.VolumeInspect(ctx, volumes, inspectOpts)
+	if err != nil {
+		return err
+	}
+	if len(volumeData) < 1 {
```

Comment:
can `len(volumeData) > 1` happen?  Do we need to raise a warning that the rest is ignored?

Reply:
@giuseppe nice catch! oh i can just drop this check, args take only 2 arguments first one is volume and second one is `input source` more than 2 args should throw error.

Reply:
It should not be able to happen, but IMO that's all the more reason to warn if it does happen